packages/eslint-plugin/src/rules/no-unnecessary-type-assertion.ts

```diff
@@ -189,14 +189,13 @@ export default createRule<Options, MessageIds>({
 
         const type = getConstrainedTypeAtLocation(services, node.expression);
 
-        if (!isNullableType(type)) {
+        if (!isNullableType(type) && !isTypeFlagSet(type, ts.TypeFlags.Void)) {
```

Comment:
question - should `void` just be part of `isNullableType`?

Reply:
@kirkwaiblinger  A quick check shows that modifying isNullableType seems to work fine. However, if we do modify this function, it would be nice to add some tests to other rules that use it. What do you think about creating a separate issue?

Reply:
Agreed!